torchtune/models/clip/_position_embeddings.py

```diff
@@ -100,23 +100,314 @@ def __init__(
+
+        Args:
+            state_dict (Dict[str, Any]): The state dict to load.
+            prefix (str): The prefix of the state dict.
+            *args (Tuple[Any]): Additional positional arguments.
+            **kwargs (Dict[str, Any]): Additional keyword arguments.
+
+        Raises:
+            ValueError: if loaded local or global embedding n_tokens_per_tile is not derived
+                from a squared grid.
+            ValueError: if after interpolation, the shape of the loaded local embedding
+                is not compatible with the current embedding.
+            ValueError: if after interpolation, the shape of the loaded global embedding
+                is not compatible with the current embedding.
+        """
+        # local
+        inpt_pos_embed = state_dict.get(prefix + "local_token_positional_embedding")
+        if inpt_pos_embed is not None:
+
+            # sanity check
+            inpt_n_tokens_per_tile, inpt_embed_dim = inpt_pos_embed.shape
+            if math.sqrt(inpt_n_tokens_per_tile - 1) % 1 != 0:
+                raise ValueError(
+                    f"Loaded local positional embedding has shape {inpt_n_tokens_per_tile=}, "
+                    f"which indicates a grid_size that is not squared. This is currently not supported."
+                )
+
+            # instantiated pos emb
+            (
+                tgt_n_tokens_per_tile,
+                tgt_embed_dim,
+            ) = self.local_token_positional_embedding.shape
+
+            # resize ckpt to match instantiated shape
+            inpt_pos_embed = self._resize_local_position_embedding(
+                inpt_pos_embed,
+                tgt_patch_grid_size=int(math.sqrt(tgt_n_tokens_per_tile - 1)),
+            )
+
+            # update state dict
+            state_dict[prefix + "local_token_positional_embedding"] = inpt_pos_embed
+            if inpt_pos_embed.shape != self.local_token_positional_embedding.shape:
+                raise ValueError(
+                    f"Loaded local positional embedding has shape {inpt_pos_embed.shape}, "
+                    f"after interpolation. Expected shape {self.local_token_positional_embedding.shape}."
+                )
+
+        # global
+        inpt_global_pos_embed = state_dict.get(
+            prefix + "global_token_positional_embedding"
+        )
+
+        if inpt_global_pos_embed is not None:
+
+            _, _, inpt_n_tokens_per_tile, _ = inpt_global_pos_embed.shape
+
+            # sanity check
+            if math.sqrt(inpt_n_tokens_per_tile - 1) % 1 != 0:
+                raise ValueError(
+                    f"Loaded local positional embedding has shape {inpt_n_tokens_per_tile=}, "
+                    f"which indicates a grid_size that is not squared. This is currently not supported."
+                )
+
+            # instantiated pos emb
+            (
+                tgt_max_num_tiles_x,
+                tgt_max_num_tiles_y,
```

Comment:
These will never differ, right? Maybe just use `tgt_max_num_tiles, _` instead?

Reply:
I thought about it, but i thought of making it explicit so the user can understand what is in this dimension. What do you think?

Reply:
Yeah no strong preference tbh, it's fine as is too. But might just add a comment somewhere, I think the usage of just `tgt_max_num_tiles_x` in L193 made me take a second glance